src/main/java/chess/domain/board/Board.java

```diff
@@ -0,0 +1,132 @@
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class Board {
+
+    private final Map<Location, Piece> board;
+
+    public Board() {
+        this.board = initialBoard();
+    }
+
+    private Map<Location, Piece> initialBoard() {
+        Map<Location, Piece> initialBoard = new HashMap<>();
+        initialPawnSetting(initialBoard);
+        initialRookSetting(initialBoard);
+        initialKnightSetting(initialBoard);
+        initialBishopSetting(initialBoard);
+        initialQueenSetting(initialBoard);
+        initialKingSetting(initialBoard);
+        return initialBoard;
+    }
+
+    private void initialPawnSetting(Map<Location, Piece> board) {
+        for (Column value : Column.values()) {
+            board.put(new Location(value, Row.TWO), new WhitePawn());
+            board.put(new Location(value, Row.SEVEN), new BlackPawn());
+        }
+    }
+
+    private void initialRookSetting(Map<Location, Piece> board) {
+        board.put(new Location(Column.A, Row.ONE), new Rook(Color.WHITE));
+        board.put(new Location(Column.A, Row.EIGHT), new Rook(Color.BLACK));
+        board.put(new Location(Column.H, Row.ONE), new Rook(Color.WHITE));
+        board.put(new Location(Column.H, Row.EIGHT), new Rook(Color.BLACK));
+    }
+
+    private void initialKnightSetting(Map<Location, Piece> board) {
+        board.put(new Location(Column.B, Row.ONE), new Knight(Color.WHITE));
+        board.put(new Location(Column.B, Row.EIGHT), new Knight(Color.BLACK));
+        board.put(new Location(Column.G, Row.ONE), new Knight(Color.WHITE));
+        board.put(new Location(Column.G, Row.EIGHT), new Knight(Color.BLACK));
+    }
+
+    private void initialBishopSetting(Map<Location, Piece> board) {
+        board.put(new Location(Column.C, Row.ONE), new Bishop(Color.WHITE));
+        board.put(new Location(Column.C, Row.EIGHT), new Bishop(Color.BLACK));
+        board.put(new Location(Column.F, Row.ONE), new Bishop(Color.WHITE));
+        board.put(new Location(Column.F, Row.EIGHT), new Bishop(Color.BLACK));
+    }
+
+    private void initialQueenSetting(Map<Location, Piece> board) {
+        board.put(new Location(Column.D, Row.ONE), new Queen(Color.WHITE));
+        board.put(new Location(Column.D, Row.EIGHT), new Queen(Color.BLACK));
+    }
+
+    private void initialKingSetting(Map<Location, Piece> board) {
+        board.put(new Location(Column.E, Row.ONE), new King(Color.WHITE));
+        board.put(new Location(Column.E, Row.EIGHT), new King(Color.BLACK));
+    }
+
+    public void move(Location source, Location target) {
+        Piece selectedPiece = findPieceAt(source);
+        Path path = createPath(source, target);
+        if (selectedPiece.canMove(path)) {
+            selectedPiece.move();
+            updateLocation(source, target, selectedPiece);
+            return;
+        }
+        throw new IllegalArgumentException("유효하지 않은 움직임입니다.");
```

Comment:
```suggestion
        if (!selectedPiece.canMove(path)) {
            throw new IllegalArgumentException("유효하지 않은 움직임입니다.");
        }
        
        selectedPiece.move();
        updateLocation(source, target, selectedPiece);
```
반환값이 없는 메서드에서는 이런 방향이 더 안정적인 코드일것 같아요. 
누군가 리팩토링하다가 return을 빼먹거나 throw 를 지워버릴 가능성이 있을것 같아서요

Reply:
뭔가 찜찜한 기분이 들었던 게 이거였나봐요!
해당 방향으로 리팩토링 해보겠습니다!